src/cheshire/core.clj

```diff
@@ -8,9 +8,34 @@
                                        JsonGenerator
                                        JsonGenerator$Feature)
            (com.fasterxml.jackson.dataformat.smile SmileFactory)
+           (cheshire.prettyprint CustomPrettyPrinter)
            (java.io StringWriter StringReader BufferedReader BufferedWriter
                     ByteArrayOutputStream OutputStream Reader Writer)))
 
+(def ^:private default-pretty-print-options
+  {:indentation 2
+   :indent-arrays? false
+   :indent-objects? true
+   :before-array-values nil
+   :after-array-values nil
+   :object-field-value-separator nil})
+
+(defn- create-pretty-printer
```

Comment:
I don't think these should be private, allow them to be created and passed instead of a map, if desired.